go/runtime/host/loadbalance/loadbalance.go

```diff
@@ -0,0 +1,300 @@
+		maybeIdx := lb.nextIdx
+		lb.nextIdx = (lb.nextIdx + 1) % len(lb.instances)
+
+		if _, faulty := lb.failedInstances[maybeIdx]; faulty {
+			continue
+		}
+
+		idx = maybeIdx
+		break
+	}
+	lb.l.Unlock()
+
+	if idx == -1 {
+		return -1, fmt.Errorf("host/loadbalance: all instances have failed")
+	}
+	return idx, nil
+}
+
+// Implements host.Runtime.
+func (lb *lbRuntime) UpdateCapabilityTEE() {
+	for _, rt := range lb.instances {
+		rt.UpdateCapabilityTEE()
+	}
+}
+
+// Implements host.Runtime.
+func (lb *lbRuntime) WatchEvents(ctx context.Context) (<-chan *host.Event, pubsub.ClosableSubscription, error) {
+	return lb.instances[0].WatchEvents(ctx)
+}
+
+// Implements host.Runtime.
+func (lb *lbRuntime) Start() {
+	lb.startOnce.Do(func() {
+		for idx, rt := range lb.instances {
+			idx := idx
+			rt := rt // Make sure goroutine below operates on the right instance.
+
+			go func() {
+				evCh, sub, err := rt.WatchEvents(context.Background())
+				if err != nil {
+					lb.logger.Error("failed to watch for instance events",
+						"err", err,
+						"instance", idx,
+					)
+					return
+				}
+				defer sub.Close()
+
+				for {
+					select {
+					case ev := <-evCh:
+						switch {
+						case ev.Started != nil:
+							// Mark instance as available.
+							lb.logger.Info("instance is available",
+								"instance", idx,
+							)
+
+							lb.l.Lock()
+							delete(lb.failedInstances, idx)
+							lb.l.Unlock()
+						case ev.FailedToStart != nil, ev.Stopped != nil:
+							// Mark instance as failed.
+							lb.logger.Warn("instance is no longer available",
+								"instance", idx,
+							)
+
+							lb.l.Lock()
+							lb.failedInstances[idx] = struct{}{}
+							lb.l.Unlock()
+						default:
+						}
+					case <-lb.stopCh:
+						return
+					}
+				}
+			}()
+
+			rt.Start()
+		}
+	})
+}
+
+// Implements host.Runtime.
+func (lb *lbRuntime) Abort(ctx context.Context, force bool) error {
+	// We don't know which instance to abort, so we abort all instances.
+	errCh := make(chan error, len(lb.instances))
+	for _, rt := range lb.instances {
+		rt := rt // Make sure goroutine below operates on the right instance.
+
+		go func() {
+			errCh <- rt.Abort(ctx, force)
+		}()
+	}
+
+	var anyErr error
+	for range lb.instances {
+		err := <-errCh
+		anyErr = errors.Join(anyErr, err)
+	}
+	return anyErr
+}
+
+// Implements host.Runtime.
+func (lb *lbRuntime) Stop() {
+	lb.stopOnce.Do(func() {
+		close(lb.stopCh)
+
+		for _, rt := range lb.instances {
+			rt.Stop()
+		}
+	})
+}
+
+type lbProvisioner struct {
+	inner host.Provisioner
+	cfg   Config
+}
+
+// Implements host.Provisioner.
+func (lb *lbProvisioner) NewRuntime(cfg host.Config) (host.Runtime, error) {
+	if lb.cfg.NumInstances < 2 {
+		// This should never happen as the provisioner constructor made sure, but just to be safe.
+		return nil, fmt.Errorf("host/loadbalance: number of instances must be at least two")
+	}
+
+	// Use the inner provisioner to provision multiple runtimes.
```

Comment:
I guess this cannot cause any problems, as we already use it to provision different versions. However, we need to be aware that attestation metrics will now be collected for all instances together.